compiler/src/dotty/tools/dotc/core/TypeEval.scala

```diff
@@ -21,6 +21,7 @@ object TypeEval:
           case tp: TypeProxy =>
             val tp1 = tp.superType
             if tp1.isStable then tp1.fixForEvaluation else tp
+          case AndType(tp1: ConstantType, tp2) if tp1.frozen_<:<(tp2) => tp1
```

Comment:
`AndType` is supposed to be commutative, so if we do this we must also do the mirror:
```suggestion
          case AndType(tp1: ConstantType, tp2) if tp1.frozen_<:<(tp2) => tp1
          case AndType(tp1, tp2: ConstantType) if tp2.frozen_<:<(tp1) => tp2
```